test/unit/proxy.test.ts

```diff
@@ -102,4 +106,107 @@ describe("proxy", () => {
     expect(resp.status).toBe(200)
     expect(await resp.json()).toBe("coder is the best")
   })
+
+  it("should handle bad requests", async () => {
+    e.use(bodyParser.json({ strict: false }))
+    e.post("/wsup", (req, res) => {
+      res.json(req.body)
+    })
+    codeServer = await integration.setup(["--auth=none"], "")
+    const resp = await codeServer.fetch(proxyPath, {
+      method: "post",
+      body: "coder is the best",
+      headers: {
+        "Content-Type": "application/json",
+      },
+    })
+    expect(resp.status).toBe(400)
+    expect(resp.statusText).toMatch("Bad Request")
+  })
+
+  it("should handle invalid routes", async () => {
+    e.post("/wsup", (req, res) => {
+      res.json(req.body)
+    })
+    codeServer = await integration.setup(["--auth=none"], "")
+    const resp = await codeServer.fetch(`${proxyPath}/hello`)
+    expect(resp.status).toBe(404)
+    expect(resp.statusText).toMatch("Not Found")
+  })
+
+  it("should handle errors", async () => {
+    e.use(bodyParser.json({ strict: false }))
+    e.post("/wsup", (req, res) => {
+      throw new Error("BROKEN")
+    })
+    codeServer = await integration.setup(["--auth=none"], "")
+    const resp = await codeServer.fetch(proxyPath, {
+      method: "post",
+      body: JSON.stringify("coder is the best"),
+      headers: {
+        "Content-Type": "application/json",
+      },
+    })
+    expect(resp.status).toBe(500)
+    expect(resp.statusText).toMatch("Internal Server Error")
+  })
+})
+
+// NOTE@jsjoeio
+// Both this test suite and the one above it are very similar
+// The main difference is this one uses http and node-fetch
+// and specifically tests the proxy in isolation vs. using
+// the httpserver abstraction we've built.
+//
+// Leaving this as a separate test suite for now because
+// we may consider refactoring the httpserver abstraction
+// in the future.
+//
+// If you're writing a test specifically for code in
+// src/node/proxy.ts, you should probably add it to
+// this test suite.
+describe("proxy (standalone)", () => {
+  const PORT = 9003
+  const PROXY_PORT = 8003
```

Comment:
❓Chose these ports at random. Not sure if there's a better way to chose two random ports.

Reply:
Could use something like https://www.npmjs.com/package/get-port but it's probably not necessary at this moment in time.